megatron/model/gpt_model.py

```diff
@@ -206,7 +206,7 @@ def universal_checkpoint_info(self):
             # Parameter that are sliced on the row dimension
             info[PARAMETER_WITH_ROW_PARALLELISM_PATTERNS] = [
                 r"\d+.mlp.dense_4h_to_h.weight",
-                r"\d+.mlp.self_attention.dense.weight",
+                r"\d+.self_attention.dense.weight",
```

Comment:
this issue that you fixed is worrisome that it was there in the first place - which means that the checks aren't strict.

perhaps this mechanism needs to be redesigned to also validate that all rules have matched, and if they haven't - assert and flag to the user that some keys/regexes are wrong.

that is while going over the regex matches, a check is needed that there was at least 1 match.

Reply:
Sounds reasonable. I will add it